src/lua/skiller/subskill_jumpstate.lua

```diff
@@ -228,20 +230,44 @@ function SubSkillJumpState:do_init()
 
    self:skill_reset()
    self.skill_status = skillstati.S_RUNNING
-
+   
    self.args = {}
+
+   -- store expected size of arguments table to compare after run of init()
+   local exp_args = {}
+   local act_args = {}
+
    for _, s in ipairs(self.skills) do
-			if s[1] ~= nil then
-				 local sname = ""
-				 if type(s[1]) == "string" then
-						sname = s[1]
-				 else
-						sname = s[1].name
-				 end
-				 self.args[sname] = {}
-			end
-	 end
+      if s[1] ~= nil then
+         local sname = ""
+         if type(s[1]) == "string" then
+            sname = s[1]
+         else
+            sname = s[1].name
+         end
+         self.args[sname] = {}
+      end
+   end
+   -- copy expected arguments table for later comparison
+   exp_args = table.deepcopy(self.args)
+
    self:init()
+   
+   -- copy actual arguments table for comparison
+   act_args = table.deepcopy(self.args)
+
+   -- compare keys of copies to determine unexpected key names
```

Comment:
This comment isn't exactly true. What the loop does is _remove_ the valid args from the `act_args` table, right? So if you rename `act_args` like I mentioned below and change this comment accordingly, the whole thing should make a lot more sense and will be much easier to understand.